torch/CMakeLists.txt

```diff
@@ -331,11 +331,11 @@ target_link_libraries(torch caffe2_library)
 
 find_package(OpenMP)
 if(OPENMP_FOUND)
-  if (VERBOSE)
-    message(STATUS "Compiling with OpenMP")
-  endif()
+  message(STATUS "Pytorch is compiling with OpenMP.")
   target_compile_options(torch INTERFACE -fopenmp)
   target_link_libraries(torch -fopenmp)
+else()
+  message(FATAL_ERROR "OpenMP library not found.")
```

Comment:
on OSX clang, OpenMP is not found

Reply:
oh, no i was just reviewing the PR.
On OSX clang, especially older clang, OpenMP is not found. so marking a FATAL_ERROR seemed dangerous.
I might have missed context though, so ignore me if I'm saying something irrelevant

Reply:
Oh, I don't know there is no OpenMP with older clang.
Maybe just use a gtest instead of a cmake FATAL_ERROR?

@ezyang @gchanan

Reply:
In https://github.com/pytorch/pytorch/issues/6133 I talk about a similar problem. I think the right thing to do is some sort of two part combo:

1. Have a test that checks that OpenMP is found and working (number of threads is greater than 1). Have this test skip by default.
2. On build configurations which we expect OpenMP to exist (I think that's all of our Linux ones), DISABLE skipping of this test via some environment variable (so that it runs and fails if OpenMP is not configured.)

The remaining problem is to ensure that there is *some* build configuration which actually turns the environment variable on; since if you rename build configurations you could accidentally delete the only build that is actually testing a configuration. I don't know of a good robust solution for this problem yet. But if we say "All Linux builds must support OpenMP" that seems reasonably robust.